pylops_mpi/basicoperators/BlockDiag.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import scipy as sp
+from mpi4py import MPI
+from typing import Optional, Sequence
+
+from pylops.utils import DTypeLike
+from pylops import LinearOperator, MatrixMult
+
+from pylops_mpi.DistributedArray import local_split, Partition
+
+sp_version = sp.__version__.split(".")
+if int(sp_version[0]) <= 1 and int(sp_version[1]) < 8:
+    from scipy.sparse.linalg.interface import _get_dtype
+else:
+    from scipy.sparse.linalg._interface import _get_dtype
+
+
+class MPIBlockDiag(LinearOperator):
+    r"""MPI Block-diagonal operator.
+
+        Create a block-diagonal operator from N linear operators using MPI.
+
+        Parameters
+        ----------
+        ops : :obj:`list`
+            Linear operators to be stacked. Alternatively,
+            :obj:`numpy.ndarray` or :obj:`scipy.sparse` matrices can be passed
+            in place of one or more operators.
+        base_comm : :obj:`mpi4py.MPI.Comm`, optional
+            Base MPI Communicator. Defaults to ``mpi4py.MPI.COMM_WORLD``.
+        dtype : :obj:`str`, optional
+            Type of elements in input array.
+
+        Attributes
+        ----------
+        shape : :obj:`tuple`
+            Operator shape
+        explicit : :obj:`bool`
+            Operator contains a matrix that can be solved explicitly (``True``) or
+            not (``False``)
+
+        Notes
+        -----
+        Refer to :obj:pylops.BlockDiag for more details.
```

Comment:
It is good to refer to pylops for the general idea of the operator, but you should add some content to explain how this specific operator works in a MPI fashion. Basically, how operators are splitted across workers, how the forward and adjoint work, etc. Try to spend more time on documentation, it is as IMPORTANT as code :)

Reply:
Updated the notes, would want your feedback...